lib/logstash/outputs/elasticsearch.rb

```diff
@@ -330,8 +337,144 @@ def register
       :max_interval => @idle_flush_time,
       :logger => @logger
     )
+
+    @retry_timer_thread = Thread.new do
+      loop do
+        sleep(@retry_max_interval)
+        @retry_queue_needs_flushing.signal
+      end
+    end
+
+    @retry_thread = Thread.new do
+      while @retry_teardown_requested.false?
+        retry_flush
+      end
+    end
```

Comment:
I think we should move the signalling logic from the `retry_flush` method to here:

```ruby
    @retry_thread = Thread.new do
      while @retry_teardown_requested.false?
        @retry_flush_mutex.synchronize{@retry_queue_needs_flushing.wait(@retry_flush_mutex)}
        retry_flush
      end
    end
```